lib/TWCManager/TWCMaster.py

```diff
@@ -164,26 +153,34 @@ def convertWattsToAmps(self, watts):
     def countSlaveTWC(self):
         return int(len(self.slaveTWCRoundRobin))
 
-    def debugLog(self, minlevel, function, message, fields = []):
+    def debugLog(self, minlevel, function, message, fields=[]):
         # Trim/pad the module name as needed
         if len(function) < 10:
             for a in range(len(function), 10):
                 function += " "
-        if self.debugOutputToFile == True:
-            self.logger.info(function + " %02d " % minlevel + message)
-
+        data = {
+            "debugLevel": self.debugLevel,
+            "fields": fields,
+            "function": function,
+            "logTime": self.time_now(),
+            "minLevel": minlevel,
+            "message": message,
+        }
+        atLeastOne = False
         # Pass the debugLog message to all enabled Logging modules
         for module in self.getModulesByType("Logging"):
-            module["ref"].debugLog(
-                {
-                    "debugLevel": self.debugLevel,
-                    "fields": fields,
-                    "function": function,
-                    "logTime": self.time_now(),
-                    "minLevel": minlevel,
-                    "message": message,
-                }
-            )
+            atLeastOne = True
+            module["ref"].debugLog(data)
+
+        # First calls won't be printed, because there is no logger registered
+        if not (atLeastOne):
+            if data["debugLevel"] >= data["minLevel"]:
+                print(
+                    colored(data["logTime"] + " ", "yellow")
+                    + colored(f("{data['function']}"), "green")
+                    + colored(f(" {data['minLevel']} "), "cyan")
+                    + f("{data['message']}")
+                )
```

Comment:
I would rather that we buffer any logs before log modules are initialized and then feed those logs back through. I'll be happy to add some code for that. The problem with this approach is that it will print out log messages in an implementation where somebody may not want this. That would then lead to a requirement for some additional configuration to mute those logs if someone wanted an entirely silent installation. Anyway, happy to take this on and fix it, no need to address it in your PR.

Reply:
So you will never have some feedback, if it fails very early. How about to introduce a "silent" config parameter instead.

Reply:
You wouldn't have some feedback today if there were an issue prior to the debugLog function being called, we're talking about the delta between when modules start getting called and the log module loads. Further to that, we'll just moved the logging modules to the top of the list to minimize disruption. The benefit of the log buffer approach is that any messages logged until the end of module initialization would go to all configured modules and not just via a single hardcoded output - currently if we log to file you won't see those entries prior to the module's initialization. The proposed way of dealing with them in your code is to print them to screen, for mine would be to buffer them and then print them to all log modules, including the configured file logging.